src/plugins/explore/public/application/utils/state_management/actions/set_dataset/set_dataset.ts

```diff
@@ -29,16 +29,24 @@ export const setDatasetActionCreator = (
     },
   } = services;
   const currentQuery = queryString.getQuery();
+
+  const state = getState();
   const {
     queryEditor: { promptModeIsAvailable },
     query,
-  } = getState();
+    ui,
+  } = state;
 
   dispatch(setActiveTab(''));
   dispatch(clearResults());
   dispatch(clearQueryStatusMap());
   dispatch(clearLastExecutedData());
 
+  const activeTab = services.tabRegistry.getTab(ui.activeTabId);
+  if (activeTab?.onInactive) {
+    activeTab?.onInactive(state);
+  }
```

Comment:
why don't we just dispatch `setPatternsField('')`?

Reply:
If we're on the logs tab or the patterns tab, and I'm switching the dataset, I would say that the tab becomes inactive. Maybe we need to clear up the definition of inactive in this case, but I see it as since the entire query and what its querying is changing, we're changing everything about the page, and it could be considered inactive, since later, it'll become active again but with a new dataset. 
So I'm just seeing it as a hard reset cycle.
The reason I didn't just dispatch `setPatternsField('')` was because of extensibility. What if later on, I need to add some more logic here other than just dispatching one redux change? Calling upon my tab's onInactive seems to open to the door for that.